apps/video/js/view.js

```diff
@@ -206,6 +208,25 @@
     dom.timeSlider.addEventListener('keypress', handleSliderKeypress);
 
     ForwardRewindController.init(dom.player, dom.seekForward, dom.seekBackward);
+
+    // Workaround to fix issue reported in Bug 1214157
+    // We would like to pause video playback when interrupted by competing audio
+    // Audio Channel API throws interruptbegin event after play (Bug 1192748)
+    // If performance.now() - lastPlayTime is small we ignore interruptbegin
+    // and pause video playback only when interrupted by competing audio.
+    // This workaround should be removed with the fix of Bug 1206581
+    dom.player.addEventListener('play', function() {
+      lastPlayTime = window.performance.now();
+    });
+
+    dom.player.addEventListener('mozinterruptbegin', function() {
+      // Ignore mozinterruptbegin event if it immediately follows play event
+      if ((window.performance.now() - lastPlayTime) > 500) {
```

Comment:
I hesitate to add a third option, since this is an alternate proposal that ideally would have been suggested earlier and this is an edge case. In any event, what if instead of using the time between 'play' and 'interrupt' we use a bool: waitingForInterruptFromPlay (or something like that), that is set to true in the 'play' event handler. In the 'interruptbegin' handler, if waitingForInterruptFromPlay is true, set it to false and return. If waitingForInterruptFromPlay is false, invoke pause. What do you think?